features/admin.server-configurations.v1/forms/password-recovery-form/password-recovery-form.tsx

```diff
@@ -0,0 +1,801 @@
+    /**
+     * Recovery link expiry time field.
+     */
+    expiryTime: string;
+    /**
+     * Sms otp expiry time field
+     */
+    smsOtpExpiryTime: string;
+    /**
+     * SMS OTP code length field.
+     */
+    smsOtpLength: string;
+    /**
+     * max resend count field.
+     */
+    maxResendCount: string;
+    /**
+     * Max allowed failed attempts count field.
+     */
+    maxFailedAttemptCount: string;
+}
+
+const allowedConnectorFields: string[] = [
+    ServerConfigurationsConstants.NOTIFY_SUCCESS,
+    ServerConfigurationsConstants.RECOVERY_LINK_EXPIRY_TIME,
+    ServerConfigurationsConstants.RECOVERY_EMAIL_LINK_ENABLE,
+    ServerConfigurationsConstants.RECOVERY_SMS_OTP_ENABLE,
+    ServerConfigurationsConstants.RECOVERY_SMS_EXPIRY_TIME,
+    ServerConfigurationsConstants.RECOVERY_OTP_USE_UPPERCASE,
+    ServerConfigurationsConstants.RECOVERY_OTP_USE_LOWERCASE,
+    ServerConfigurationsConstants.RECOVERY_OTP_USE_NUMERIC,
+    ServerConfigurationsConstants.RECOVERY_OTP_LENGTH,
+    ServerConfigurationsConstants.RECOVERY_MAX_RESEND_COUNT,
+    ServerConfigurationsConstants.RECOVERY_MAX_FAILED_ATTEMPTS_COUNT
+];
+
+const FORM_ID: string = "governance-connectors-password-recovery-form";
+
+/**
+ * Password Recovery Configuration Form.
+ *
+ * @param props - Props injected to the component.
+ * @returns Functional component.
+ */
+export const PasswordRecoveryConfigurationForm: FunctionComponent<PasswordRecoveryConfigurationFormPropsInterface> = (
+    props: PasswordRecoveryConfigurationFormPropsInterface
+): ReactElement => {
+
+    const {
+        initialValues,
+        onSubmit,
+        readOnly,
+        isConnectorEnabled,
+        isSubmitting,
+        ["data-testid"]: testId
+    } = props;
+
+    const { t } = useTranslation();
+    const [ initialConnectorValues, setInitialConnectorValues ]
+        = useState<PasswordRecoveryFormInitialValuesInterface>(undefined);
+    const [ isEmailRecoveryEnabled, setIsEmailRecoveryEnabled ] = useState<boolean>(false);
+    const [ isSMSRecoveryEnabled, setIsSMSRecoveryEnabled ] = useState<boolean>(false);
+    const [ isUpperCaseEnabled, setIsUpperCaseEnabled ] = useState<boolean>(false);
+    const [ isLowerCaseEnabled, setIsLowerCaseEnabled ] = useState<boolean>(false);
+    const [ isNumericEnabled, setIsNumericEnabled ] = useState<boolean>(false);
+    const showSmsOtpPwdRecoveryFeatureStatusChip: boolean =
+        useSelector((state: AppState) => state?.config?.ui?.showSmsOtpPwdRecoveryFeatureStatusChip);
+
+    /**
+     * Flattens and resolved form initial values and field metadata.
+     */
+    useEffect(() => {
+        if (isEmpty(initialValues?.properties)) {
+            return;
+        }
+
+        let resolvedInitialValues: PasswordRecoveryFormInitialValuesInterface = null;
+
+        initialValues.properties.map((property: ConnectorPropertyInterface) => {
+            if (allowedConnectorFields.includes(property.name)) {
```

Comment:
change this to a switch

Reply:
Addressed with https://github.com/wso2/identity-apps/pull/6418/commits/0dd77c8c16b11b96a91e6b1331294e3bb8560a5b